### PR TITLE
feat(FX-4025): Add active filter pills to other (non-artist) artwork grids

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR solves [FX-4025]

Review app: https://pills-for-other-grids.artsy.net/

Investigation and faced problems: [Spike Document](https://www.notion.so/artsy/Spike-Add-saved-search-alerts-to-other-artwork-grids-a57a04ac02654861915f6f7d3f2edb4a)

### Added support for grids
* [Auctions](https://pills-for-other-grids.artsy.net/auction/artsy-x-seoul-auction-contemporary-icons)
* [Show](https://pills-for-other-grids.artsy.net/show/charlie-smith-london-papyrophilia)
* [Tag](https://pills-for-other-grids.artsy.net/tag/cowboy)
* [Artist Series](https://pills-for-other-grids.artsy.net/artist-series/hiroshi-sugimoto-seascapes)
* [Gene](https://pills-for-other-grids.artsy.net/gene/ukiyo-e)
* [Fair](https://pills-for-other-grids.artsy.net/fair/art-cologne-2020/artworks)
* [Collection](https://pills-for-other-grids.artsy.net/collection/contemporary)
* [Search Results](https://pills-for-other-grids.artsy.net/search?term=sculpture)
* Collect
  * [Default collect page](https://pills-for-other-grids.artsy.net/collect)
  * [Collect page with selected medium value by default](https://pills-for-other-grids.artsy.net/collect/sculpture)

### Description

[FX-4025]: https://artsyproduct.atlassian.net/browse/FX-4025?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Components/SavedSearchAlert/Utils/extractPills.ts [hunk 1]
```diff
@@ -31,18 +35,24 @@ export const extractPillFromAggregation = (
...
   const { paramName, paramValue } = filter
   const aggregation = aggregationForFilter(paramName, aggregations)
+  let aggregationValues = aggregation?.counts
 
-  if (aggregation) {
-    const aggregationByValue = keyBy(aggregation.counts, "value")
+  // Use hardcoded medium values for some grids (e.g. fair, collect grids)
+  if (paramName === "additionalGeneIDs" && !aggregation) {
+    aggregationValues = hardcodedMediums as Aggregation["counts"]
+  }
```
Comment (dimatretyak) [inline src/v2/Components/SavedSearchAlert/Utils/extractPills.ts:43]: You can read more about this moment [here](https://www.notion.so/artsy/Spike-Add-saved-search-alerts-to-other-artwork-grids-a57a04ac02654861915f6f7d3f2edb4a#951a63b67d384ecb84dcb8895ede1e9e)

**Comment by dimatretyak** [comment]
Please note that the `collect/:medium` pages display a non-removable default pill, since the same value is displayed as selected in filters by default and this value cannot be unselected

#### Image
demo

#### Video
https://user-images.githubusercontent.com/3513494/177790857-1589e951-b4cb-4e9c-92f1-9c28a12be2ba.mp4

File: src/v2/Apps/Auction/Components/AuctionArtworkFilter.tsx [hunk 2]
```diff
@@ -53,6 +54,7 @@ const AuctionArtworkFilter: React.FC<AuctionArtworkFilterProps> = ({
             <MediumFilter expanded />
           </>
         }
+        FilterPillsSection={<ActiveFilterPills />}
```
Comment (mdole) [inline src/v2/Apps/Auction/Components/AuctionArtworkFilter.tsx:57]: I noticed that `Medium` pills on auction pages would render with no name - I believe they are hardcoded in the same way as /collect and need the same treatment
Screen Shot 2022-07-08 at 11 23 53
!

**Review by mdole — CHANGES_REQUESTED** [review]
Wow, amazing to see this come together - and for it to be so little code (as a result of all the nice refactors you worked on @dimatretyak!)

Two things that I noticed and that I think we should fix before merging:
- All pages: When user changes the price range slider and then dismisses the price filter via the pill, the slider doesn't reset (see screen recording)
- Auction pages: Medium filters don't have names in the pills

Otherwise looking great to me!

https://user-images.githubusercontent.com/5361806/177962457-d4a4c945-7cc0-4ac1-9b98-00ff55b7431a.mov

File: src/v2/Apps/Auction/Components/AuctionArtworkFilter.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/v2/Apps/Auction/Components/AuctionArtworkFilter.tsx:57]: hmm, good catch. I'll take a look

**Comment by dimatretyak** [comment]
> All pages: When user changes the price range slider and then dismisses the price filter via the pill, the slider doesn't reset (see screen recording)

After [recent changes](https://github.com/artsy/force/pull/10425), there was also a problem that if the user enters values in inputs, the changes will not be displayed on the slider. I will add fixes in a separate PR

https://user-images.githubusercontent.com/3513494/177977870-3dd2c94c-618c-460a-8a61-03e25a7747ce.mp4

File: src/v2/Apps/Auction/Components/AuctionArtworkFilter.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/v2/Apps/Auction/Components/AuctionArtworkFilter.tsx:57]: Fixed

**Review by mdole — APPROVED** [review]
Addressing the price filter changes in a separate PR sounds good to me! Thanks for the quick fix @dimatretyak !

**Comment by dimatretyak** [comment]
> Addressing the price filter changes in a separate PR sounds good to me! 

Fixed here: https://github.com/artsy/palette/pull/1192

**Review by gkartalis — APPROVED** [review]

**Pull request merged**